server/app/routes/lecture.js

```diff
@@ -50,16 +60,43 @@ router.post('/', function(req, res, next) {
 
 router.put('/:lectureId', function(req, res, next) {
   var io = req.app.get('socketio');
+
   req.lecture.update(req.body)
   .then(function(updatedLecture) {
     if (req.body.startTime && !req.body.endTime) {
       io.emit('startLecture', updatedLecture);
+      res.status(201).json(updatedLecture);
     }
     else if (req.body.endTime) {
       io.emit('endLecture', updatedLecture);
+      res.status(201).json(updatedLecture);
+
+      Question.findAll({where: {lectureId: req.lecture.id}})
```

Comment:
The logic here looks kinda heavy. I'd recommend shifting it over to the model (e.g. to an instance method called `.endLecture`).